watcher.go

```diff
@@ -147,7 +151,9 @@ func (w *Watcher) WatchVaultToken(token string) error {
 			return errors.Wrap(err, "watcher")
 		}
 		// fakeNotifier is defined near end of file
-		w.Register(fakeNotifier(vaultTokenDummyTemplateID), vt)
+		n := fakeNotifier(vaultTokenDummyTemplateID)
+		w.Register(n)
```

Comment:
Just wanted to make sure I understood here. `w.Register` was updated to return an error. I was wondering if you were considering checking the error here now that it is available? I might be misunderstanding the intention of Register()

Reply:
Register can return an error, but it requires a setup that makes it easy to not trigger. It is designed more for code where you might be registering templates in different spots, to make sure you don't double register. In tests you have a fresh watcher with only those Register calls you just made right there.

TLDR; The error is checked in the tests that exercise that functionality, but it would just be a lot of line noise in the other tests.

Reply:
Lol... ignore that. I was just looking at tests and was thinking that was in a test as well. It is not. I'll look at it more closely.

Reply:
That probably should check the returned error there. I was thinking it not being an issue as you'd only ever need one of those vault token watcher, but you certainly could call that method twice by mistake...

Reply:
fixed.

Reply:
:) I gotcha!